### PR TITLE
http: adding another smuggling test

### PR DESCRIPTION
**Pull request by alyssawilk**
Making sure T-E + C-L checks handle tab based whitespace.

Risk Level: n/e (test only)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/integration/integration_test.cc [hunk 1]
```diff
@@ -521,6 +521,14 @@ TEST_P(IntegrationTest, TestSmuggling) {
     sendRawHttpAndWaitForResponse(lookupPort("http"), full_request.c_str(), &response, false);
     EXPECT_THAT(response, StartsWith("HTTP/1.1 400 Bad Request\r\n"));
   }
+  {
+    std::string response;
+    const std::string full_request = "GET / HTTP/1.1\r\nHost: host\r\ncontent-length: "
+                                     "36\r\n\ttransfer-encoding: chunked\r\n\r\n" +
```
Comment (antoniovicente) [inline test/integration/integration_test.cc:527]: Would it be possible to break this request string at \r\n boundaries?

Something like:
```
"GET / HTTP/1.1\r\n"
"Host: host\r\n"
"content-length: 36\r\n"
"\ttransfer-encoding: chunked\r\n\r\n"
```

It took me a while to figure out that the difference between this request and the one above is the \t just before transfer-encoding.  A comment about this being about \t would be helpful.

Also, consider a test of transfer-encoding with \t before it but no content-length, and \t before content-length and no transfer-encoding header.

File: test/integration/integration_test.cc [hunk 2]
```diff
@@ -521,6 +521,14 @@ TEST_P(IntegrationTest, TestSmuggling) {
     sendRawHttpAndWaitForResponse(lookupPort("http"), full_request.c_str(), &response, false);
     EXPECT_THAT(response, StartsWith("HTTP/1.1 400 Bad Request\r\n"));
   }
+  {
+    std::string response;
+    const std::string full_request = "GET / HTTP/1.1\r\nHost: host\r\ncontent-length: "
```
Comment (antoniovicente) [inline test/integration/integration_test.cc:526]: I think that in order to smuggle a request you need either content-length: 0 and possibly a request body that is chunk encoded prior to the smuggled request.  A 400 here could be due to the request not being chunk encoded despite there being a transfer-encoding: chunked header.

**Review by antoniovicente — COMMENTED** [review]
Thanks for the extra tests for smuggling cases.

File: test/integration/integration_test.cc [hunk 3]
```diff
@@ -521,6 +521,14 @@ TEST_P(IntegrationTest, TestSmuggling) {
     sendRawHttpAndWaitForResponse(lookupPort("http"), full_request.c_str(), &response, false);
     EXPECT_THAT(response, StartsWith("HTTP/1.1 400 Bad Request\r\n"));
   }
+  {
+    std::string response;
```
Comment (antoniovicente) [inline test/integration/integration_test.cc:525]: Also, do we have smuggling tests that use HTTP methods that can have body; for example POST.

Comment (alyssawilk) [reply test/integration/integration_test.cc:525]: GET can have body in Envoy :-/

**Review by antoniovicente — APPROVED** [review]
Thanks for the extra tests.

**Pull request merged**